### PR TITLE
Decouple Donation from DonationOrClaim

### PR DESCRIPTION
**Pull request by rahulrajaram**
Decouple Donation from DonationOrClaim, and DonorDashboardScreen from DashboardScreen to facilitate easier independent code-development of Donation and Claim and DonorDashboard and ClientDashboard. Please see commit messages for details.

This commit does not introduce any functional changes in the client or the donor apps.

-------

Testing: I've verified that existing Client, and Donor dashboards are preserved.

### CODE REVIEW

File: src/screens/DonorDashboardScreen/DonorDonationScreen/DonorDonationScreen.type.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export type Claim = {
```
Comment (jacksonrya) [inline src/screens/DonorDashboardScreen/DonorDonationScreen/DonorDonationScreen.type.ts:1]: More type duplicates?

File: src/screens/DashboardScreen/DashboardScreen.tsx [hunk 2]
```diff
@@ -4,14 +4,20 @@ import { View, Text } from 'react-native';
...
+import DonationsOrClaims from '../../library/DonationsOrClaims';
 import styles from './DashboardScreen.styles';
+import DonorDashboardScreen from '../DonorDashboardScreen';
 
 const DashboardScreen = () => {
 	const { navigate } = useNavigation();
 	const [ state ] = useGlobal();
 	const { userIdentity } = state;
-	const title = userIdentity === 'donor' ? 'My Donations.' : 'Open Donations.';
+
```
Comment (jacksonrya) [inline src/screens/DashboardScreen/DashboardScreen.tsx:15]: I like this. I wonder, for the future, if this Screen should just act as a conditional in pointing to either `DonorDashboardScreen` or a potential `ClientDashboardScreen`

Comment (rahulrajaram) [reply src/screens/DashboardScreen/DashboardScreen.tsx:15]: Yes, a near-future plan should be to create a well-defined `ClientDashboardScreen`.

**Comment by rahulrajaram** [comment]
@jacksonrya , I've fixed the linting errors. Can you merge this PR?

**Comment by jacksonrya** [comment]
@rahulrajaram for sure

PS heads up, slack call right now if u want to join. (i'll delete this comment in an edit :) )

**Pull request merged**